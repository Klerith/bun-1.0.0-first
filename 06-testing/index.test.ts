import { describe, expect, test, beforeAll } from "bun:test";



beforeAll(() => {
  // setup tests
});

describe("math", () => {

  
  test("addition", () => {
    expect(2 + 2).not.toBe(5);
    expect(7 + 3).toMatchSnapshot();
  });

  
  test('test objects snapshot', () => {

    const person = {
      name: 'John',
      age: 33,
    }

    expect(person).toMatchSnapshot();

  })


  test('test objects, expect data', () => {


    const person = {
      name: 'John',
      age: 33,
      id: '12873612873akjsd'
    }

    expect(person).toEqual({
      name: 'John',
      age: 33,
      id: expect.any(String),
    })


  })



});


