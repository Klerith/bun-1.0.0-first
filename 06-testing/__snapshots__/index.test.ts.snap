// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`math addition 1`] = `10`;

exports[`math test objects snapshot 1`] = `
{
  "age": 33,
  "name": "John",
}
`;
